src/librustdoc/formats/renderer.rs

```diff
@@ -91,20 +91,20 @@ crate fn run_format<'tcx, T: FormatRenderer<'tcx>>(
             }
             let _timer = prof.generic_activity_with_arg("render_mod_item", name.as_str());
 
-            cx.mod_item_in(&item, &name)?;
+            format_renderer.mod_item_in(&item, &name)?;
             let module = match *item.kind {
                 clean::StrippedItem(box clean::ModuleItem(m)) | clean::ModuleItem(m) => m,
                 _ => unreachable!(),
             };
             for it in module.items {
                 debug!("Adding {:?} to worklist", it.name);
-                work.push((cx.clone(), it));
+                work.push(it);
             }
 
-            cx.mod_item_out(&name)?;
+            format_renderer.mod_item_out(&name)?;
         } else if item.name.is_some() {
             prof.generic_activity_with_arg("render_item", &*item.name.unwrap_or(unknown).as_str())
-                .run(|| cx.item(item))?;
+                .run(|| format_renderer.item(item))?;
         }
     }
```

Comment:
Why does the code work this way, where it constantly clones the `Context`?

Reply:
Because some `DocContext` fields are `rc`, meaning they're not clone so to speak. ;)

Reply:
Hm, so maybe a way to clean this up could be to make `SharedContext` the only thing that's actually shared?